parsec/core/gui/new_version.py

```diff
@@ -47,28 +55,32 @@ def _fetch_json_releases():
         elif current_arch == "i386":
             win_version = "win32"
         else:
-            return (latest_from_head, url)
+            return latest_from_head, url
 
-        latest_version = (0, 0, 0)
+        latest_version = Version("0.0.0")
         latest_url = ""
 
-        for release in json_releases:
-            try:
+        try:
```

Comment:
Why the switch between try and for ? given the except does a return I don't see the difference...

Reply:
As it was finally decided to make a return at the first json inconsistency, as it means that data can't be trusted and we should redirect the user to the latest version in any case, we might as well put the iteration on `json_releases` in the try block

Reply:
Also, it covers the highly unlikely case where `json_releases` is not iterable

Reply:
👍 

(note we should also add a `except  JSONDecodeError` in `_fetch_json_releases` to protect against json inconsistency)

Reply:
Also done